### PR TITLE
feat: add gaming sdk api

### PR DESCRIPTION
**Pull request by luu-alex**
milestone 2
adds chainsafe gaming API call to plugin.
https://www.notion.so/chainsafe/Web3js-Marketplace-Plugin-e52134db8223409b9003323924cd1b3a

### CODE REVIEW

**Comment by luu-alex** [comment]
tests failing due to insufficient funds, might need to manually run contract tests instead of having it a part of the ci.

File: src/marketplacePlugin.ts [hunk 1]
```diff
@@ -140,6 +145,102 @@ export class Web3MarketplacePlugin extends Web3PluginBase {
...
+	 * @param {string} marketplaceId - Marketplace ID to query.
+	 * @param {string} customProjectId - Project ID to query (optional).
+	 * @returns {Promise<marketplaceItemsResult>} - Returns the marketplace items response object.
+	 */
+	public async getMarketplaceItems(
+		marketplaceId: string,
+		customProjectId: string,
+	): Promise<marketplaceItemsResult> {
+		// Define the API base URL
+		const baseUrl = 'https://api.gaming.chainsafe.io/v1';
```
Comment (jdevcs) [inline src/marketplacePlugin.ts]: I'll suggest to make baseurl as class member and set it via constructor. also have 'https://api.gaming.chainsafe.io/v1' default value of that via constructor, as its used in multiple places in this class.

File: src/marketplacePlugin.ts [hunk 2]
```diff
@@ -140,6 +145,102 @@ export class Web3MarketplacePlugin extends Web3PluginBase {
...
+
+		// Construct the full URL with path and optional project ID as a query parameter
+		const url = `${baseUrl}/projects/${customProjectId}/marketplaces/${marketplaceId}/items`;
+
+		// Send the GET request to the API
+		const response = await fetch(url);
+
+		// Check if the response is OK (status code 200-299)
+		if (!response.ok) {
+			throw new Error(`Failed to fetch marketplace items: ${response.statusText}`);
```
Comment (jdevcs) [inline src/marketplacePlugin.ts]: For all Errors in this plugin, lets follow Errors defining pattern as we have in web3.

File: src/types.ts [hunk 3]
```diff
@@ -1,4 +1,66 @@
...
+
+export type marketplaceItemsResult = {
+	page_number: number;
+	page_size: number;
+	total: number;
+	cursor: string;
+	items: marketplaceItem[];
+};
+
+export type Owner = {
```
Comment (danforbes) [inline src/types.ts:55]: Why do some types start with a capital letter and others start with a lowercase letter?

**Review by danforbes — APPROVED** [review]
Looks good to me, thanks for adding good docs 🙏🏻 Just one question about naming consistency.

**Review by jdevcs — APPROVED** [review]

File: src/types.ts (hunk 3 above)
Comment (luu-alex) [reply src/types.ts:55]: will update this, good point

**Pull request merged**